prometheus_client/core.py

```diff
@@ -181,6 +181,10 @@ def __eq__(self, other):
                 self.type == other.type and
                 self.samples == other.samples)
 
+    def __repr__(self):
```

Comment:
A __str__ would make more sense I think

Reply:
renamed it to `__str__` and having `repr` call `str` because it's useful for comparing unit test output:

```
E       First differing element 0:
E       <Metric name: a, documentation: help, type: counter, samples: [(u'a', {u'foo': u'bar'}, 1), (u'a', {u'foo': u'baez'}, 2), (u'a', {u'foo': u'buz'}, 3)]>
E       <Metric name: a, documentation: help, type: counter, samples: [(u'a', {u'foo': u'bar'}, 1.0), (u'a', {u'foo': u'baz'}, 2.0), (u'a', {u'foo': u'buz'}, 3.0)]>
E
E       - [<Metric name: a, documentation: help, type: counter, samples: [(u'a', {u'foo': u'bar'}, 1), (u'a', {u'foo': u'baez'}, 2), (u'a', {u'foo': u'buz'}, 3)]>]
E       ?                                                                                                                  -
E
E       + [<Metric name: a, documentation: help, type: counter, samples: [(u'a', {u'foo': u'bar'}, 1.0), (u'a', {u'foo': u'baz'}, 2.0), (u'a', {u'foo': u'buz'}, 3.0)]>]
E       ?
```
instead of 
```
E       First differing element 0:
E       <prometheus_client.core.CounterMetricFamily object at 0x108018c50>
E       <prometheus_client.core.Metric object at 0x108018cd0>
E
E       - [<prometheus_client.core.CounterMetricFamily object at 0x108018c50>]
E       ?                          -------      ------                    ^
E
E       + [<prometheus_client.core.Metric object at 0x108018cd0>]
```

Reply:
Usually str calls repr rather than the other way around. repr should usually be an instantiatable version of object, while str is more human readable.

Reply:
Modified `repr` to fit what it's supposed to be, still way more readable for tests outputs 👍